eslint-plugin-prettier.js

```diff
@@ -0,0 +1,306 @@
+/**
+ * @fileoverview Runs `prettier` as an ESLint rule.
+ * @author Andres Suarez
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+//  Requirements
+// ------------------------------------------------------------------------------
+
+const diff = require('fast-diff');
+const docblock = require('jest-docblock');
+
+// ------------------------------------------------------------------------------
+//  Constants
+// ------------------------------------------------------------------------------
+
+// Preferred Facebook style.
+const FB_PRETTIER_OPTIONS = {
+  singleQuote: true,
+  trailingComma: 'all',
+  bracketSpacing: false,
+  jsxBracketSameLine: true,
+  parser: 'flow'
+};
+
+const LINE_ENDING_RE = /\r\n|[\r\n\u2028\u2029]/g;
+
+// ------------------------------------------------------------------------------
+//  Privates
+// ------------------------------------------------------------------------------
+
+// Lazily-loaded Prettier.
+let prettier;
+
+// ------------------------------------------------------------------------------
+//  Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * Gets the location of a given index in the source code for a given context.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {number} index - An index in the source code.
+ * @returns {Object} An object containing numeric `line` and `column` keys.
+ */
+function getLocFromIndex(context, index) {
+  // If `sourceCode.getLocFromIndex` is available from ESLint, use it.
+  // Otherwise, use the private version from eslint/lib/ast-utils.
+  // `sourceCode.getLocFromIndex` was added in ESLint 3.16.0.
+  const sourceCode = context.getSourceCode();
+  if (typeof sourceCode.getLocFromIndex === 'function') {
+    return sourceCode.getLocFromIndex(index);
+  }
+  const astUtils = require('eslint/lib/ast-utils');
+  return astUtils.getLocationFromRangeIndex(sourceCode, index);
+}
+
+/**
+ * Converts invisible characters to a commonly recognizable visible form.
+ * @param {string} str - The string with invisibles to convert.
+ * @returns {string} The converted string.
+ */
+function showInvisibles(str) {
+  let ret = '';
+  for (let i = 0; i < str.length; i++) {
+    switch (str[i]) {
+      case ' ':
+        ret += '\u00B7'; // Middle Dot
+        break;
+      case '\n':
+        ret += '\u23ce'; // Return Symbol
+        break;
+      case '\t':
+        ret += '\u21b9'; // Left Arrow To Bar Over Right Arrow To Bar
```

Comment:
Nitpick: Maybe this would be more readable if you used the symbols themselves in the code rather than their escape codes. For example:

```js
switch (str[i]) {
  case ' ':
    ret += '·'; // Middle Dot, \u00b7
    break;
  case '\n':
    ret += '⏎'; // Return Symbol, \u23ce
  case '\t':
    ret += '↹'; // Left Arrow To Bar Over Right Arrow To Bar, \u21b9
}
```

That way, if someone sees a symbol in the output, it's easier to tell where in the code the symbol is coming from.